fingerprint/infra/nec-bio-sdk/src/main/java/com/simprints/fingerprint/infra/necsdkimpl/initialization/SdkInitializerImpl.kt

```diff
@@ -15,16 +18,23 @@ internal class SdkInitializerImpl @Inject constructor(
     @ApplicationContext private val context: Context,
     private val necInstance: NEC,
     private val licenseRepository: LicenseRepository,
+    private val saveLicenseCheck: SaveLicenseCheckEventUseCase
 ) : SdkInitializer<Unit> {
     override suspend fun initialize(initializationParams: Unit?) {
-        val licence = licenseRepository.getCachedLicense(Vendor.NEC)
+        var licenseStatus: LicenseStatus? = null
         try {
-            necInstance.init(licence.encodeAndConvertToByteBuffer(), context)
+            val licence = licenseRepository.getCachedLicense(Vendor.NEC)
+            licenseStatus = licence.determineLicenseStatus()
+            if (licenseStatus != LicenseStatus.VALID) {
+                throw BioSdkException.BioSdkInitializationException(message = "License is $licenseStatus")
+            }
+            necInstance.init(licence!!.data.encodeAndConvertToByteBuffer(), context)
```

Comment:
I'd advocate against using the `!!` operator unless absolutely necessary. Is it possible to avoid this?